bin/hrx

```diff
@@ -67,28 +75,37 @@ class HRX::CLI < Thor
     end
 
     hrx = HRX::Archive.load(archive)
-    files_extracted = hrx.entries.count do |entry|
-      destination = File.join(directory, entry.path)
-      if entry.is_a?(HRX::Directory)
-        FileUtils.mkdir_p(destination)
-        false
-      else
-        FileUtils.mkdir_p(File.dirname(destination))
-        if !File.exist?(destination) ||
-           file_collision(destination) {entry.content}
-          File.write(destination, entry.content)
-          true
+    if options[:level]
+      if options[:level] < 1
+        raise Thor::Error.new("--level must be greater than or equal to 2.")
+      elsif !options[:level].is_a?(Integer)
+        raise Thor::Error.new("--level must be an integer.")
+      end
+
+      files_extracted = hrx.entries.group_by do |e|
+        dir = File.dirname(e.path)
+        dir == "." ? [] : dir.split("/")[0...options[:level]]
+      end.sum do |(components, entries)|
+        if components.length < options[:level]
+          _write_entries(directory, entries)
         else
-          false
+          path = components.join("/")
+          if _write("#{directory}/#{path}.hrx", hrx.child_archive(path).to_hrx)
```

Comment:
```suggestion
          _write("#{directory}/#{path}.hrx", hrx.child_archive(path).to_hrx) ? 1 : 0
```

nit: Maybe? Since we're already 3 levels deep, it would be nice to avoid another level of if/else nesting.

Reply:
I thought about this, but I think it ends up hiding the fact that there's a return value here, since the condition takes up so much of the horizontal space and is mostly called for the side effect. I'll split the difference by assigning it to a variable and using a ternary operator on that.